### PR TITLE
Create Blog “meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche”

### PR DESCRIPTION
**Pull request by GanaelLaplanche**
Automatically generated by Netlify CMS

### CODE REVIEW

**Review by drensing — COMMENTED** [review]
Ganael,
Thank you so much for your most enjoyable blog post. As the HPE Developer Community blog editor, it's my job to check for readability of the posts and suggest alternative phrasing where it may help. Please take my edits as the simple suggestions that they are. If you have any questions or need anything clarified, feel free to reach out!

All the best,
Dale

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+title: Meet Open Source enthusiast and Fpart project developer, Ganael Laplanche
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md:2]: Hello Dale, Thank you very much for your suggestions, I'll answer below...

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
...tarted when I was working for a renowned center for biomedical research after a discussion with a friend of mine. We wanted to implement a fast bin-packing tool to produce filesystem tree partitions with the same size and number of files. The tool quickly evolved and got support for hooks that can be triggered when a partition is generated. At that time, we needed to move petabyte-scale filesystems to freshly-acquired storage arrays. With its new hooking system, fpart seemed to be a good basement to launch small migration jobs in parallel through our SSH cluster. Initial tests ([see our article in french](https://connect.ed-diamond.com/GNU-Linux-Magazine/glmf-164/parallelisez-vos-transferts-de-fichiers)) were successful but we were still depending on our on-site scheduler to orchestrate submitted jobs and it was to be retired soon or later. We needed a new scheduler. That's where [fpsync](https://www.fpart.org/fpsync/) comes into play : the tools wraps fpart and embeds its own scheduler to trigger small rsync jobs to parallelize data migration *by itself*. It can leverage your SSH cluster to get the best from your data servers, acting as a powerful -standalone- data migration tool.
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: I've followed those recommendations, thanks!

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Of course, as an ardent open source supporter, those tools were released with an open source license (BSD 2-Clause "Simplified" License). They quickly became used by big companies (Intel, AWS, Microsoft, Alibaba, Oracle, ...) as well as research centers to migrate petabyte-scale filesystems.
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md:26]: Fpsync is a single tool, the plural was a typo in the previous paragraph :)

I've modified the second sentence, thanks!

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+# What attracted you to free software?
+
+I first discovered free software reading magazines (trying GNU/Linux distros offered on CDROM) surfing on Linux hype during mid-90's. But I really understood what free software meant later during my studies. I was immediately seduced as that single concept gathers human beings' best : sharing knowledge to move forward together. As a student, the most important aspect has been that one : being able to *learn*, as the code is freely available and the open source community very responsive. But I then quickly felt I owed something to that community : I didn't want to use all that free (as in free beer) software without giving something back. So I started looking at how I could contribute back. This is where FreeBSD played a important role, acting as a catalyzer...
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: I've mostly followed your suggestions except for two things : the magazines were surfing on Linux hype, not me ;-) Also, I did go to a private high school, not to University. I've applied other changes, thanks!

[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: (so I've rephrased the magazines part, trying to make it clearer)

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+About development, specifically, FreeBSD is a good choice because it is POSIX compliant ; this is important if you want to produce portable code. Also, it is very easy to access source code for world, kernel and ports (third-party applications ported to FreeBSD). One can easily patch things and test the modifications, which is a bit harder on other systems where you would often have to install a dedicated source package to be able to patch it.
+
+Finally, the system is a pleasure to administrate and update. I think I have not reinstalled my machine since late 2000's, just performing upgrades since. Third-party applications can be easily installed and upgraded using binary packages, which was not the case when I started using FreeBSD.
+
+All those reasons are why I use FreeBSD on my systems, for servers and development but also as a daily desktop OS. Lots of people still think FreeBSD is not ready for everyday use on a desktop, I am living proof this is not true!
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: Modified, thanks!

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+# What other open source projects are you involved with?
+
+I became a FreeBSD developer in 2011 and I now maintain more than 40 ports (a port is a set of patches and build options that makes a software work on FreeBSD, it also acts as the basis for binary packages). Maintaining ports is a fantastic hobby because one the one hand you have the chance to work for your favorite OS, and on the other hand you can contribute patches back upstream, so you are always in relation with different communities.
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: Yes, this was a typo (one -> on). I've modified the sentences, thanks!

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+Aside from my FreeBSD activities, I have more [personal projects](https://contribs.martymac.org/): I wrote about Fpart and Fpsync, but I am also the author of the ldapscripts, a set of tools to simplify user and group management within an LDAP directory. They are quite old now, but they still do the job. I also worked on various smaller projects, such as sms1xxx kernel module (a port of Linux' Siano DVB-T driver to FreeBSD, now deprecated in favor of webcamd), evtViewer (a viewer for Ms event log files) or Grpar (a Build engine group archive extract tool). I also wrote several courses (in french).
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: Modified, thanks!

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+I also try to contribute to software I use when I find a bug (either by fixing it or at least by reporting it).
+
+# Is there anything else you’d like to share with our readers?
+
+I owe a lot to free software. That's mostly what allowed me to learn computing, that's what made my career possible. That's why I contribute back the most I can.
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: Modified, thanks!

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+But that takes time (that is, personal time) and money (we need machines to test on, as well as power to run them). I am glad to see more and more companies supporting open source. Recently, HPE gave me a replacement for my old server, I'll never thank them enough for that kindness! This is a sign that things are changing: this would probably have not been possible a few years ago. I think everybody now understands why it is so important to support open source development. Giving hardware is a kind of support, sharing code is another one. Let's encourage companies to continue that way!
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: Frederic suggested me a change in that paragraph that I have added.

I've added the following sentence : "This HPE ProLiant ML350 allows me to build my ports far more quicker than with my old machine."

Is that enough?

Thanks for your very nice and useful feedback :)

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+# Why did you come to FreeBSD as a development platform?
+
+There are several reasons for that choice. As a curious student, I first tried [FreeBSD](https://www.freebsd.org/) in early 2000's, testing version 4.5. What impressed me first at that time was its documentation (["handbook"](https://docs.freebsd.org/en/books/handbook/)) and man pages. While GNU/Linux appeared to be complex to me, FreeBSD suddently appeared to be clear. With a very nice and welcoming community, it was the perfect platform for a newcomer into the UNIX world. I got caught and never returned to other systems since...
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: "man" pages is short for "manual" pages :) I've added that precision and applied your suggested changes, thanks!

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+I later understood another reason of that clarity : it is a homogeneous system, not a patchwork of very different projects. That makes all the difference: a specific version of FreeBSD represents a specific version of world (a selection of base components) *and* kernel, making a complete system. Those components are maintained by the same entity ([FreeBSD developers](https://docs.freebsd.org/en/articles/contributors/)) and, as such, everything is consistent, from options to documentation to man pages. This is a great value for users and a guarantee of robustness and stability for the system.
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md:38]: I wanted to stress that a specific version of FreeBSD is a specific version of (world + kernel) couple. So I rephrased that again a little bit. Don't hesitate to tell me if this is not clear (or not written in good English :p).

As I defined "man" pages before, I let that expression as is, considering the reader now knows what that means.

Regarding the "world", I think you understood what it means :)

Thanks !

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+About development, specifically, FreeBSD is a good choice because it is POSIX compliant ; this is important if you want to produce portable code. Also, it is very easy to access source code for world, kernel and ports (third-party applications ported to FreeBSD). One can easily patch things and test the modifications, which is a bit harder on other systems where you would often have to install a dedicated source package to be able to patch it.
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: Change applied, thanks!

File: content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+
+Finally, the system is a pleasure to administrate and update. I think I have not reinstalled my machine since late 2000's, just performing upgrades since. Third-party applications can be easily installed and upgraded using binary packages, which was not the case when I started using FreeBSD.
```
[the remark this answers is not in the document]
Comment (GanaelLaplanche) [reply content/blog/meet-open-source-contributor-and-fpart-fpsync-project-developer-ganael-laplanche.md]: I've followed your suggestions, thanks!

**Pull request merged**